esp32_s2_kaluga_kit/include/bsp/esp32_s2_kaluga_kit.h

```diff
@@ -74,83 +74,45 @@
 extern "C" {
 #endif
 
-/**************************************************************************************************
- *
- * I2S audio interface
- *
- * There is one device connected to the I2S peripheral:
- *  - Codec ES8311 (plyback and recording)
- **************************************************************************************************/
-
 /**
- * @brief Kaluga-kit I2S pinout
+ * @brief BSP display configuration structure
  *
- * Can be used for i2s_std_gpio_config_t and/or i2s_std_config_t initialization
  */
-#define BSP_I2S_GPIO_CFG       \
-    {                          \
-        .mclk = BSP_I2S_MCLK,  \
-        .bclk = BSP_I2S_SCLK,  \
-        .ws = BSP_I2S_LCLK,    \
-        .dout = BSP_I2S_DOUT,  \
-        .din = BSP_I2S_DSIN,   \
-        .invert_flags = {      \
-            .mclk_inv = false, \
-            .bclk_inv = false, \
-            .ws_inv = false,   \
-        },                     \
-    }
+typedef struct {
+    lvgl_port_cfg_t lvgl_port_cfg;
+} bsp_display_cfg_t;
 
-/**
- * @brief Mono Duplex I2S configuration structure
+/**************************************************************************************************
  *
- * This configuration is used by default in bsp_audio_init()
- */
-#define BSP_I2S_DUPLEX_MONO_CFG(_sample_rate)                                                         \
-    {                                                                                                 \
-        .clk_cfg = I2S_STD_CLK_DEFAULT_CONFIG(_sample_rate),                                          \
-        .slot_cfg = I2S_STD_PHILIP_SLOT_DEFAULT_CONFIG(I2S_DATA_BIT_WIDTH_16BIT, I2S_SLOT_MODE_MONO), \
-        .gpio_cfg = BSP_I2S_GPIO_CFG,                                                                 \
-    }
-
-/**
- * @brief ES8311 init structure
+ * I2S audio interface
  *
- */
-#define BSP_ES8311_SCLK_CONFIG(_sample_rate) \
-    {                                        \
-        .mclk_from_mclk_pin = false,         \
-        .mclk_inverted = false,              \
-        .sclk_inverted = false,              \
-        .sample_frequency = _sample_rate     \
-    }
+ * There is one device connected to the I2S peripheral:
+ *  - Codec ES8311 (playback and recording)
+ *
+ * After speaker initialization, use functions from esp_codec_dev for play/record audio.
+ * Example audio play:
+ * \code{.c}
+ * esp_codec_dev_handle_t spk_codec_dev = bsp_audio_codec_speaker_init();
+ * esp_codec_dev_set_out_vol(spk_codec_dev, DEFAULT_VOLUME);
+ * esp_codec_dev_open(spk_codec_dev, &fs);
+ * esp_codec_dev_write(spk_codec_dev, wav_bytes, wav_bytes_len);
+ * esp_codec_dev_close(spk_codec_dev);
+ * \endcode
+ **************************************************************************************************/
 
 /**
- * @brief Init audio
+ * @brief Initialize speaker codec device
```

Comment:
When I compare it with another boards, here are missing two functions:

```
#if ESP_IDF_VERSION < ESP_IDF_VERSION_VAL(5, 0, 0)
esp_err_t bsp_audio_init(const i2s_config_t *i2s_config);
#else
esp_err_t bsp_audio_init(const i2s_std_config_t *i2s_config);
#endif
```

```
const audio_codec_data_if_t *bsp_audio_get_codec_itf(void);

```

Can we make it same?

Reply:
IMHO, having different API for v4.4 and v5.x breaks the whole idea of backward compatibility.
But it is probably too late for this discussion... OK, will update